Platforms/Export-Import-Platform.ps1

```diff
@@ -110,136 +129,214 @@ Function Join-ExceptionMessage
+	} catch {
 		Write-Host -ForegroundColor Red $_.Exception.Response.StatusDescription
 		$logonToken = ""
 	}
-    If ($logonToken -eq "")
-    {
-        Write-Host -ForegroundColor Red "Logon Token is Empty - Cannot login"
-        return
-    }
+	If ($logonToken -eq "") {
+		Write-Host -ForegroundColor Red "Logon Token is Empty - Cannot login"
+		return
+	}
 	
-    # Create a Logon Token Header (This will be used through out all the script)
-    # ---------------------------
-    $logonHeader =  New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
-    $logonHeader.Add("Authorization", $logonToken)
-#endregion
+	# Create a Logon Token Header (This will be used through out all the script)
+	# ---------------------------
+	$logonHeader = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
+	$logonHeader.Add("Authorization", $logonToken)
+	#endregion
+
+	switch ($PsCmdlet.ParameterSetName) {
+		"Import" {
+			import-platform $PlatformZipPath -error
+		}
+
+		"ImportFile" {
+			foreach ($line in Get-Content $listFile) {
+				if ("" -ne $line) { import-platform $line }
+			} 
+		}
 
-	switch($PsCmdlet.ParameterSetName)
-	{
-		"Import"
-		{
-			If (Test-Path $PlatformZipPath)
-			{
-				$zipContent = [System.IO.File]::ReadAllBytes($(Resolve-Path $PlatformZipPath))
-				$importBody = @{ ImportFile=$zipContent; } | ConvertTo-Json -Depth 3 -Compress
-				try{
-					$ImportPlatformResponse = Invoke-RestMethod -Method POST -Uri $URL_ImportPlatforms -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700 -Body $importBody
-					Write-Debug "Platform ID imported: $($ImportPlatformResponse.PlatformID)"
-					Write-Host "Retrieving Platform details"
-					# Get the Platform Name
-					$platformDetails = Invoke-RestMethod -Method Get -Uri $($URL_PlatformDetails -f $ImportPlatformResponse.PlatformID) -Headers $logonHeader -ContentType "application/json" -TimeoutSec 2700
-					If($platformDetails)
-					{
-						Write-Debug $platformDetails
-						Write-Host "$($platformDetails.Details.PolicyName) (ID: $($platformDetails.PlatformID)) was successfully imported and $(if($platformDetails.Active) { "Activated" } else { "Inactive" })"
-						Write-Host "Platform details:" 
-						$platformDetails.Details | Select-Object PolicyID, AllowedSafes, AllowManualChange, PerformPeriodicChange, @{Name = 'AllowManualVerification'; Expression = { $_.VFAllowManualVerification}}, @{Name = 'PerformPeriodicVerification'; Expression = { $_.VFPerformPeriodicVerification}}, @{Name = 'AllowManualReconciliation'; Expression = { $_.RCAllowManualReconciliation}}, @{Name = 'PerformAutoReconcileWhenUnsynced'; Expression = { $_.RCAutomaticReconcileWhenUnsynched}}, PasswordLength, MinUpperCase, MinLowerCase, MinDigit, MinSpecial 
-					}		
-				} catch {
-					#Write-Error $_.Exception
-					Write-Error $_.Exception.Response
-					Write-Error $_.Exception.Response.StatusDescription
-				}
-			}
+		"Export" {
+			export-platform $PlatformID
```

Comment:
Thinking about it, maybe we should verify that Platform Id is not empty
It is mandatory, but still can be " " or an invalid character for example